python/mdvtools/tests/test_stress_performance.py

```
@@ -0,0 +1,455 @@
+        factory = MockAnnDataFactory(random_seed=42)
+        
+        # Force garbage collection
+        gc.collect()
+        initial_memory = get_memory_usage()
+        
+        for i in range(3):  # Reduced from 5 to 3 for larger datasets
+            # Create new dataset each time with smaller size
+            adata = factory.create_realistic(5000, 2000)
+            
+            with temp_mdv_project() as test_dir:
+                with suppress_anndata_warnings():
```

Ruff (0.11.9)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)